central/graphql/resolvers/image_components.go

```diff
@@ -335,13 +339,37 @@ func (resolver *imageComponentResolver) ImageVulnerabilityCounter(ctx context.Co
 
 func (resolver *imageComponentResolver) ImageVulnerabilities(ctx context.Context, args PaginatedQuery) ([]ImageVulnerabilityResolver, error) {
 	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "ImageVulnerabilities")
-	return resolver.root.ImageVulnerabilities(resolver.withImageComponentScope(ctx), args)
+
+	// Short path. Full image is embedded when image scan resolver is called.
+	embeddedComponent := embeddedobjs.ComponentFromContext(resolver.ctx)
+	if embeddedComponent == nil {
+		return resolver.root.ImageVulnerabilities(resolver.withImageComponentScope(ctx), args)
+	}
+
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return nil, err
+	}
+
+	pagination := query.GetPagination()
+	query.Pagination = nil
```

Comment:
why is pagination removed here? getImageCVEResolvers should respect the pagination